brainscore_vision/model_helpers/activations/temporal/core/extractor.py

```diff
@@ -0,0 +1,235 @@
+        # This is done here, before storing, so that we only store the reduced activations.
+        reduced_paths = self._reduce_paths(stimuli_paths)
+        activations = fnc(layers=layers, stimuli_paths=reduced_paths)
+        activations = self._expand_paths(activations, original_paths=stimuli_paths)
+        return activations
+
+    @store_xarray(identifier_ignore=['stimuli_paths', 'layers'], combine_fields={'layers': 'layer'})
+    def _from_paths_stored(self, identifier, layers, stimuli_identifier, stimuli_paths):
+        stimuli_paths.sort()
+        return self._from_paths(layers=layers, stimuli_paths=stimuli_paths)
+
+    def _from_paths(self, layers, stimuli_paths):
+        if len(layers) == 0:
+            raise ValueError("No layers passed to retrieve activations from")
+        return self.inferencer(stimuli_paths, layers)
+
+    def _reduce_paths(self, stimuli_paths):
+        return list(set(stimuli_paths))
+
+    def _expand_paths(self, activations, original_paths):
+        activations_paths = activations['stimulus_path'].values
+        argsort_indices = np.argsort(activations_paths)
+        sorted_x = activations_paths[argsort_indices]
+        sorted_index = np.searchsorted(sorted_x, original_paths)
+        index = [argsort_indices[i] for i in sorted_index]
+        return activations[{'stimulus_path': index}]
+
+    def register_batch_activations_hook(self, hook):
+        r"""
+        The hook will be called every time a batch of activations is retrieved.
+        The hook should have the following signature::
+
+            hook(batch_activations) -> batch_activations
+
+        The hook should return new batch_activations which will be used in place of the previous ones.
+        """
+
+        handle = HookHandle(self._batch_activations_hooks)
+        self._batch_activations_hooks[handle.id] = hook
+        return handle
+
+    def register_stimulus_set_hook(self, hook):
+        r"""
+        The hook will be called every time before a stimulus set is processed.
+        The hook should have the following signature::
+
+            hook(stimulus_set) -> stimulus_set
+
+        The hook should return a new stimulus_set which will be used in place of the previous one.
+        """
+
+        handle = HookHandle(self._stimulus_set_hooks)
+        self._stimulus_set_hooks[handle.id] = hook
+        return handle
+
+
+def change_dict(d, change_function, keep_name=False, multithread=False):
+    if not multithread:
+        map_fnc = map
+    else:
+        pool = ThreadPool()
+        map_fnc = pool.map
+
+    def apply_change(layer_values):
+        layer, values = layer_values
+        values = change_function(values) if not keep_name else change_function(layer, values)
+        return layer, values
+
+    results = map_fnc(apply_change, d.items())
+    results = OrderedDict(results)
+    if multithread:
+        pool.close()
+    return results
+
+
+def lstrip_local(path):
```

Comment:
typehints, comments (I realize we never commented in the original code, but this is a good chance to improve)